src/python/dxpy/templating/templates/nextflow/src/nextflow.sh

```diff
@@ -110,86 +242,129 @@ dx_path() {
-      set -x && env | grep -v DX_SECURITY_CONTEXT | sort
-    fi
-    
-    if [ -n "$docker_creds" ]; then
-        dx-registry-login
-    fi
+  if [[ $debug == true ]]; then
+    export NXF_DEBUG=2
+    TRACE_CMD="-trace nextflow.plugin"
+    env | grep -v DX_SECURITY_CONTEXT | sort
+    set -x
+  fi
+
+  DX_CACHEDIR=$DX_PROJECT_CONTEXT_ID:/.nextflow_cache_db
+  validate_run_opts
+
+  if [[ $preserve_cache == true ]]; then
+    check_cache_db_storage
+  fi
+
+  if [ -n "$docker_creds" ]; then
+    dx-registry-login
+  fi
+
+  # set default NXF env constants
+  export NXF_DOCKER_LEGACY=true
+  #export NXF_DOCKER_CREDS_FILE=$docker_creds_file
+  #[[ $scm_file ]] && export NXF_SCM_FILE=$(dx_path $scm_file 'Nextflow CSM file')
+  export NXF_HOME=/opt/nextflow
+  export NXF_ANSI_LOG=false
+  export NXF_PLUGINS_DEFAULT=nextaur@1.1.0
+  export NXF_EXECUTOR='dnanexus'
+
+  # use /home/dnanexus/nextflow_playground as the temporary nextflow execution folder
+  mkdir -p /home/dnanexus/nextflow_playground
+  cd /home/dnanexus/nextflow_playground
+
+  # get job output destination
+  DX_JOB_OUTDIR=$(jq -r '[.project, .folder] | join(":")' /home/dnanexus/dnanexus-job.json)
+  # initiate log file
+  LOG_NAME="nextflow-$DX_JOB_ID.log"
+
+  # get current executable name
+  EXECUTABLE_NAME=$(jq -r .executableName /home/dnanexus/dnanexus-job.json)
+
+  # set/create current session id
+  if [[ -n $resume ]]; then
+    get_resume_session_id
+  else
+    NXF_UUID=$(uuidgen)
+  fi
+  export NXF_UUID
+  export NXF_CACHE_MODE=LENIENT
 
-    DX_WORK=${work_dir:-$DX_WORKSPACE_ID:/scratch/}
-
-    LOG_NAME="nextflow-$(date +"%y%m%d-%H%M%S").log"
-    # parse dnanexus-job.json to get job output destination
-    OUT_PROJECT=$(jq -r .project /home/dnanexus/dnanexus-job.json)
-    OUT_FOLDER=$(jq -r .folder /home/dnanexus/dnanexus-job.json)
-    OUTDIR="$OUT_PROJECT:${OUT_FOLDER#/}"
-    DX_LOG=${log_file:-"$OUTDIR/$LOG_NAME"}
-
-    export NXF_WORK=dx://$DX_WORK
-    export NXF_HOME=/opt/nextflow
-    export NXF_UUID=$(uuidgen)
-    export NXF_ANSI_LOG=false
-    export NXF_EXECUTOR=dnanexus
-    export NXF_PLUGINS_DEFAULT=nextaur@1.1.0
-    export NXF_DOCKER_LEGACY=true
-    #export NXF_DOCKER_CREDS_FILE=$docker_creds_file
-    #[[ $scm_file ]] && export NXF_SCM_FILE=$(dx_path $scm_file 'Nextflow CSM file')
-
-    mkdir -p /home/dnanexus/out/output_files
-    cd /home/dnanexus/out/output_files
-    
-    generate_runtime_config
-    
-    NEXTFLOW_CMD="nextflow \
-      ${TRACE_CMD} \
-      $nextflow_top_level_opts \
-      ${RUNTIME_CONFIG} \
-      -log ${LOG_NAME} \
-      run @@RESOURCES_SUBPATH@@ \
-      @@PROFILE_ARG@@ \
-      -name run-${NXF_UUID} \
-      $nextflow_run_opts \
-      $nextflow_pipeline_params \
-      @@REQUIRED_RUNTIME_PARAMS@@
+  if [[ $preserve_cache == true ]]; then
+    dx set_properties "$DX_JOB_ID" nextflow_executable="$EXECUTABLE_NAME" nextflow_session_id="$NXF_UUID" nextflow_preserve_cache="$preserve_cache"
```

Comment:
A minor suggestion (for readability), not necessary to merge if requires additional testing

```suggestion
    dx set_properties "$DX_JOB_ID" \
            nextflow_executable="$EXECUTABLE_NAME" \
            nextflow_session_id="$NXF_UUID" \
            nextflow_preserve_cache="$preserve_cache"
```